src/fsharp/TypeChecker.fs

```diff
@@ -504,8 +504,8 @@ type cenv =
       /// Used to resolve names
       nameResolver: NameResolver
       
-      /// The set of active conditional defines
-      conditionalDefines: string list
+      /// The set of active conditional defines. The value is None when conditional erasure is disabled in tooling.
+      conditionalDefines: string list option
```

Comment:
I'm not a fan of `list option` types as I usually get confused by what it could mean. The comment does help though, so it's not a mystery. Though, I prefer another `bool` field honestly. What you think?

Reply:
Why are you ok with it? (Just curious because I'm on the fence).

The more I think about it, the more I really want a new bool field i.e `isConditionalErasureEnabled : bool`. Nothing can be clearer than that especially when it's used in code.

Using an `option` gives the field `conditionalDefines` more responsibility.